erc777-syntax.k

```diff
@@ -6,40 +6,63 @@ module ERC777-SYNTAX
   imports BOOL-SYNTAX
 
   //
-  // Token API
+  // ERC777Token
   //
 
-  syntax Exp ::= name ( )                              // String
-               | symbol ( )                            // String
-               | totalSupply ( )                       // Number
-               | balanceOf ( Address )                 // Number
-               | granularity ( )                       // Number
-               | defaultOperators ( )                  // Addresses
-               | isOperatorFor ( Address , Address )   // Bool
-
-  syntax Cmd ::= authorizeOperator ( Address )
-               | revokeOperator ( Address )
-               | send ( Address , Number , Data )
-               | operatorSend ( Address , Address , Number , Data , Data )
-               | burn ( Number , Data )
-               | operatorBurn ( Address , Number , Data , Data )
+  syntax Exp ::= AExp | BExp | Set
+
+  syntax Exp ::= name ( )
+               | symbol ( )
+
+  syntax AExp ::= totalSupply ( )
+                | balanceOf ( AExp ) [strict]
+                | granularity ( )
+
+  syntax Exp ::= defaultOperators ( )
+
+  syntax BExp ::= isOperatorFor ( AExp , AExp ) [strict]
+
+  syntax Exp ::=  authorizeOperator ( AExp )                        [strict]
+                | revokeOperator ( AExp )                           [strict]
+                | send ( AExp , AExp , Data )                       [strict]
+                | operatorSend ( AExp , AExp , AExp , Data , Data ) [strict]
+                | burn ( AExp )                                     [strict]
+                | operatorBurn ( AExp , AExp , Data )               [strict]
```

Comment:
I know that you followed the way in which the erc20 semantics is defined here, but i'm not happy with the use of AExp, especially because it doesn't capture the difference between Address and Number.